department-of-veterans-affairs/abd-vro
# Feature - use logging to generate api metrics

## What was the problem?

Make response time for `/classifier` endpoint available in datadog

Associated tickets or Slack threads:
- Previously, we discussed the option to make use of the datadog api client in [379](https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/379)
- Instead, log-based metrics are used in order to circumvent issues w/ partner team deploy to DEV
- follow-up in [443](https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/443) to include the log messages as metrics and create a dashboard widget

## How does this fix it?[^1]

- new logs are added via middleware

## How to test this PR
- run API locally and make request to /classifier endpoint
- observe response time in logs


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
